alhimick-prog/hometask-4/my_array_rotate.rb

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# Extension for native class Array
+class Array
+  def lab_rotate(shift = 1)
+    return self unless check_before_rotate?(self, shift)
+
+    itself_rotate(self, shift)
+  end
+
+  def lab_rotate!(shift = 1)
+    return self unless check_before_rotate?(self, shift)
+
+    replace(itself_rotate(self, shift))
+    self
```

Comment:
@alhimick-prog do we really need this?

Reply:
@lincAlinc do you mean `self`? Yes, this is not needed.